lib/__init__.py

```diff
@@ -130,7 +130,7 @@ def connect(dsn=None, connection_factory=None, cursor_factory=None, **kwargs):
         conn = getConnectionBalanced(lbprops, connection_factory, cursor_factory, **kwasync)
         if conn != None:
             return conn
-        print('Failed to apply load balancing, Trying normal connection')
+        raise psycopg2.OperationalError("Could not find a server to connect to.")
```

Comment:
As discussed, it is not required since the driver would have already tried to connect to all the eligible host (as per the connection properties set) and would only come here if none of the eligible hosts are available to connect.